src/internal_events/sample.rs

```diff
@@ -7,5 +9,9 @@ pub struct SampleEventDiscarded;
 impl InternalEvent for SampleEventDiscarded {
     fn emit(self) {
         counter!("events_discarded_total", 1);
+        emit!(ComponentEventsDropped::<INTENTIONAL> {
+            count: 1,
+            reason: "sample discarded"
+        })
```

Comment:
Same suggestion as in https://github.com/vectordotdev/vector/pull/14770#discussion_r992397906 to add a comment to deprecate the other internal events above